### PR TITLE
Make tensor parallelism functional on HPU

### PR DESCRIPTION
**Pull request by tkrupa-intel**
Functional on 1.15-# 341.

### CODE REVIEW

**Comment by tkrupa-intel** [comment]
@kzawora-intel Created PR based on the rebased version, please close the other one.

File: vllm/engine/llm_engine.py [hunk 1]
```diff
@@ -158,15 +162,17 @@ def _init_workers_ray(self, placement_group: "PlacementGroup",
...
                 continue
             if self.parallel_config.tensor_parallel_size == 1:
                 num_gpus = self.cache_config.gpu_memory_utilization
             else:
                 num_gpus = 1
             worker = ray.remote(
                 num_cpus=0,
-                num_gpus=num_gpus,
+                num_gpus=num_gpus if not is_hpu() else 0,
+                resources={'HPU': num_gpus} if is_hpu else {},
```
Comment (mdvoretc-intel) [inline vllm/engine/llm_engine.py:175]:
```suggestion
                resources={'HPU': num_gpus} if is_hpu() else {},
```

File: vllm/utils.py [hunk 2]
```diff
@@ -35,13 +35,11 @@ def is_hpu() -> bool:
...
-    from vllm.hpu import cuda_utils
-else:
-    from vllm._C import cuda_utils
-
-
 def get_max_shared_memory_bytes(gpu: int = 0) -> int:
+    if is_hpu():
+        from vllm.hpu import cuda_utils
+    else:
+        from vllm._C import cuda_utils
```
Comment (mdvoretc-intel) [inline vllm/utils.py:42]: What is the purpose of this change?

File: vllm/worker/worker.py [hunk 3]
```diff
@@ -62,10 +62,14 @@ def init_model(self) -> None:
...
-        self.device = torch.device(f"cuda:{local_rank}")
+        device_str = f"cuda:{local_rank}" if not is_hpu() else "hpu"
+        self.device = torch.device(device_str)
         if self.rank < 0:
             raise ValueError("Invalid or unspecified rank.")
-        torch.cuda.set_device(self.device)
+        if is_hpu():
+            torch.hpu.set_device(self.device)
+        else:
+            torch.cuda.set_device(self.device)
```
Comment (mdvoretc-intel) [inline vllm/worker/worker.py:72]: Are the changes here carried over from the huda decoupling, or is there a specific need to make this worker change for this PR?

Comment (kzawora-intel) [reply vllm/worker/worker.py:72]: It is needed specifically for this PR.

**Review by mdvoretc-intel — APPROVED** [review]
